MovieEnthusiast.Infrastructure/Persistence/Repositories/MovieRepository.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.EntityFrameworkCore;
+using MovieEnthusiast.Application.Common.Interfaces;
+using MovieEnthusiast.Application.Common.Models;
+using MovieEnthusiast.Domain.Entities;
+
+namespace MovieEnthusiast.Infrastructure.Persistence.Repositories;
+
+public class MovieRepository : IMovieRepository
+{
+    private readonly ApplicationDbContext _context;
+
+    public MovieRepository(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<PaginatedEntity<Movie>> GetMovies(
+        Pagination pagination,
+        CancellationToken cancellationToken)
+    {
+        var position = (pagination.PageNumber - 1) * pagination.PageSize;
+
+        var movies = await _context.Movies
```

Comment:
Please add `.AsNoTracking()` as we are not updating the returned entities and in query below
[ref](https://learn.microsoft.com/en-us/ef/core/querying/tracking#no-tracking-queries)